docs/contributing/how-integration-tests-work.md

```diff
@@ -0,0 +1,45 @@
+---
+title: "How integration tests work"
+linkTitle: "How integration tests work"
+weight: 5
+slug: how-integration-tests-work
+---
+
+Cortex integration tests are written in Go and based on a [custom framework](https://github.com/cortexproject/cortex/tree/master/integration/e2e) running Cortex and its dependencies in Docker containers and using the Go [`testing`](https://golang.org/pkg/testing/) package for assertions. Integration tests run in CI for every PR, and can be easily executed locally during development (it just requires Docker).
+
+## How to run integration tests
+
+When integration tests run in CI, we build the Cortex docker image based on the PR code and then run the integration tests against it. On the contrary, when running tests **locally** you should first build the Cortex Docker image:
+
+```
+make ./cmd/cortex/.uptodate
```

Comment:
I just run `make`, and that builds images.

Reply:
Yes, but it's just slower because builds multiple images even the one not strictly required by integration tests. Thoughts?

Reply:
That's good to know. I think I'll create an alias for this. Thanks